src/helpers/request.ts

```diff
@@ -41,16 +41,6 @@ export const createRequestHandler = <Args>(
   };
 };
 
-/**
- * Variant of `createRequestHandler` used when the args object is entirely optional.
- * We cannot combine this with `createRequestHandler` because it is impossible to make the function
- * parameter conditionally nullable or non-nullable.
- */
-export const createRequestHandlerOptional = <Args>(
-  fn: (a?: Args) => BaseRequestParams,
-): ((a?: Args, additionalFetchOptions?: AdditionalPerFetchParams) => CompleteRequestParams) =>
-  createRequestHandler(fn);
```

Comment:
This didn't work because `initMakeRequest` is still annotating the type of the outputted function using `HandleRequest<Args>`. It's too much pain to get the types flowing nicely so I'll skip it for now. It's a non-breaking change to bring it back in later.